### ISSUE TITLE
[java] AvoidDeeplyNestedIfStmts violations can be unintentionally undetected

### ISSUE THREAD

**Issue by HoushCE29** [pmd/pmd#1499]
**Affects PMD Version:** 5.3.3, 6.*, 7.0.0

**Rule:** [design/AvoidDeeplyNestedIfStatements](https://pmd.github.io/pmd-6.9.0/pmd_rules_java_design.html#avoiddeeplynestedifstmts)

**Description:** This rule has strange behavior where if you dispatch from an else block from somewhere else to a method containing violating deeply nested if code, it doesn't get caught as a violation. But if you do a simple if block but no else block, then it gets caught as a violation.

It's hard to explain without writing a book.... so see the sample code...

**Code Sample demonstrating the issue:**

This is a sample violating method:
```java
private String doSomething(String somethingToDo) {
    if (condition) {
        if (anotherCondition) {
            if (problemDepthCondition) {
                return SOMETHING;
            }
        }
    }
    return SOMETHING_ELSE;
}
```

Here is code that the violation is caught for:
```java
public String performSomething(String action) {
    if (action.equals(NOTHING)) {
        return NOTHING;
    }
    return doSomething(action);
}
```

However, this code (which is effectively the same) allows the violation:
```java
public String performSomething(String action) {
    if (action.equals(NOTHING)) {
        return NOTHING;
    }
    // Only difference: using an else block
    else {
        return doSomething(action);
    }
}
```

Sure, the nested if's _can_ and _should_ be avoided and rewritten, but I would expect that if someone did code that way (which is what happened and why I am bringing up this issue) that it would be caught, even in the `else` block...

Please let me know if this actually isn't an issue (and is expected behavior), or if further clarification is needed.

**Running PMD through:** *Gradle*

**Comment by HoushCE29** [comment]
@jsotuyod Will do! I'll get back to you in the next week.

**Comment by HoushCE29** [comment]
@jsotuyod Confirmed, it does in fact do that same behavior with version 6.9.0.

Do you require a copy of my "test-case" mini Gradle project for further evaluation?

**Comment by adangel** [comment]
It seems to be related to https://sourceforge.net/p/pmd/bugs/44/ - I was wondering, whether we always excluded somehow the else-branch. And we do this pretty long now (since 2002 - 7e4cdbd77ab00e607f1e87f66e5db082caa44d13).
I agree, that adding an else, should not silence the rule...
